src/strategies/MangroveOffer.sol

```diff
@@ -134,23 +132,15 @@ abstract contract MangroveOffer is AccessControlled, IOfferLogic {
 
   /// @inheritdoc IOfferLogic
   function reserve(address maker) public view override returns (address) {
-    address reserve_ = MOS.getStorage().reserves[maker];
-    return reserve_ == address(0) ? maker : reserve_;
+    return __reserve__(maker);
   }
 
-  /// @inheritdoc IOfferLogic
-  function setReserve(address maker, address reserve_) public override onlyCaller(maker) {
-    require(__checkReserveApproval__(reserve_, maker), "mgvOffer/makerNotApproved");
-    MOS.getStorage().reserves[maker] = reserve_;
-    emit SetReserve(maker, reserve_);
+  /// @notice hook to customize offer owner's reserve for the offer logic
+  /// @param maker the offer owner's address whose address is being queried
+  function __reserve__(address maker) internal view virtual returns (address) {
+    return maker;
   }
 
-  /// @notice verifies that maker is allowed to use a reserve for pooling funds
-  /// @param reserve_ the reserve on which one is pooling funds
-  /// @param maker the pooler of the reserve
-  /// @dev function throws if `reserve_` has not approved `maker`
-  function __checkReserveApproval__(address reserve_, address maker) internal virtual returns (bool);
-
```

Comment:
The `__reserve__` hook replaces the `reserves` mapping in storage. Default returned value is the offer maker's address.
`__checkReserveApproval__` is no longer needed, as reserve is no longer (necessarily) controlled by the offer maker (simply the location where her funds are to be pull from and placed to.